src/tray.rs

```diff
@@ -1,11 +1,53 @@
+use std::io::Cursor;
+
+use image::Rgba;
 use ksni;
 
 #[derive(Debug)]
-struct Tray;
+struct Tray {
+    icon: ksni::Icon,
+}
+
+impl Tray {
+    pub fn new() -> Self {
+        let rgba_image = {
+            let cursor = {
+                let image_bytes = include_bytes!("../assets/flakeshot_logo_dpi_96.png");
+                Cursor::new(image_bytes.to_owned())
+            };
+            image::io::Reader::with_format(cursor, image::ImageFormat::Png)
+                .decode()
+                .unwrap()
+                .to_rgba8()
+        };
+
+        let (width, height) = rgba_image.dimensions();
+
+        let data = rgba_image
+            .pixels()
+            // rgba => argb
+            .map(|pixel| Rgba::from([pixel[3], pixel[0], pixel[1], pixel[2]]))
+            .fold(
+                Vec::with_capacity((width * height) as usize),
+                |mut prev, pixel| {
+                    prev.extend_from_slice(&pixel.0);
+                    prev
+                },
+            );
```

Comment:
```suggestion
        let mut data: Vec<u8> = rgba_image.into_vec();
        
        // rgba => argb
        for i in (0..data.len()).step_by(4) {
            data[i..i + 4].rotate_right(1);
        }

```

or if you like the functional approcah

```suggestion
        let mut data: Vec<u8> = rgba_image.into_vec();

        // rgba => argb
        data.chunks_mut(4).for_each(|pixel| pixel.rotate_right(1));
```

Reply:
that's a good catch!